Doc/whatsnew/3.15.rst

```diff
@@ -75,6 +75,10 @@ New features
 Other language changes
 ======================
 
+* Python UTF-8 mode is now enabled by default.
+  It may be disabled with by setting :envvar:`PYTHONUTF8=0 <PYTHONUTF8>` as
+  an environment variable or by using the :option:`-X utf8=0 <-X>` flag.
+  See :pep:`686` for further details.
```

Comment:
I feel like we can probably put some more explanation in here, such as that it affects TextIOWrapper and hence `open()`. The current description doesn't sound as scary as it needs to, in my opinion.

Along the lines of: "Python UTF-8 mode is now enabled by default. This means that (files/console/etc.) will now use UTF-8 regardless of system settings, unless specifically overridden in code (typically with an ``encoding=`` argument)."

Reply:
To be clear, it's nothing new. But we shouldn't assume that everyone already knows what UTF-8 mode implies. There are many more people out there who haven't ever thought about it than those who are waiting for it to be the default.

Reply:
Another effect of the UTF-8 Mode is that Python ignores the locale encoding.

Reply:
I've expanded the *What's New* entry, please see https://cpython-previews--133712.org.readthedocs.build/en/133712/whatsnew/3.15.html#other-language-changes